Added an example

yc2454 opened the pull request:
I added a new example of ssm in C, simpleadd.c, which is a small program that adds two numbers together.

Commit:
added an example

j-hui commented on simpleadd.c:
Nit: add a newline at the end of the file, as is standard for text files on UNIX systems (you should be able to configure your text editor to do this automatically!)

j-hui commented on simpleadd.c:
Nit: initialize these SVs with `DEBUG_SV_NAME`, to attach human-readable names to each declared variable for debugging purposes. We're not debuggin this but it's good to be principled, since the compiler will be generating this code for us.

j-hui reviewed (changes requested):
You need to add your example to the `Makefile`, so that it builds when I run `make`. I'll add a change on the `sched` branch to the `Makefile` that should make this slightly easier.

Furthermore, this program doesn't actually terminate correctly (and leaks memory because of it). I think there's a bug in the `step_add`. Can you fix it?

j-hui commented:
(Oh and once you try to compile the file, you will also find a type error. That should be pretty straightforward to fix too.)

yc2454 committed:
Merge branch 'sedwards-lab:sched' into sched

Commit:
made some changes

Commit:
got rid of warnings

j-hui committed:
Suppress formatting warning on Linux

Note that this leads to a formatting warning when built on macOS,
because for some reason uint64_t is typedef'd to unsigned long long
rather than unsigned long.

j-hui committed:
Fix formatting

j-hui committed:
Add new example target to gitignore

j-hui reviewed (approved):
Looks great, thanks for making those changes. I'll merge this in.
